#!/usr/bin/env python
PACKAGE = "rc_tagdetect_client"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("use_cached_images", bool_t, 0, "Use most recently received image pair instead of waiting for a new pair", False)
gen.add("forget_after_n_detections", int_t, 0, "Number of detection runs after which to forget about a previous tag during tag re-identification", 30, 1, 1000)
gen.add("max_corner_distance", double_t, 0, "Maximum distance of corresponding tag corners in meters during tag re-identification", 0.005, 0.001, 0.01)
gen.add("quality", str_t, 0, "Quality of tag detection (H, M or L)", "H")
gen.add("detect_inverted_tags", bool_t, 0, "Detect tags with black and white exchanged", False)

gen.add("publish_visualization", bool_t, 0, "Whether or not the tf and markers should be published", True)


exit(gen.generate(PACKAGE, "rc_tagdetect_client", "TagDetect"))
